src/main/groovy/com/sequenceiq/ambari/client/services/BlueprintService.groovy

```diff
@@ -208,26 +207,50 @@ trait BlueprintService extends ClusterService {
    */
   def String extendBlueprintHostGroupConfiguration(String blueprintJson, Map<String, Map<String, Map<String, String>>> newConfigs) {
     def blueprintMap = slurper.parseText(blueprintJson)
-    for (int j = 0; j < newConfigs.size(); j++) {
-      def configurations = blueprintMap.host_groups.find { it.name == newConfigs.keySet().getAt(j) }.configurations
+    def globalConfigs = !blueprintMap.configurations ? [:] : blueprintMap.configurations
+    def filteredNewConfigs = [:]
+    newConfigs.each {
+      def hostgroup = it.key
+      def hostgroupConfig = it.value
+      def filteredHostgroupConfigs = [:]
+      filteredNewConfigs.put(hostgroup, filteredHostgroupConfigs)
+      hostgroupConfig.each {
+        def hostgroupSite = it.key
+        def siteMap = it.value
+        def filteredSiteMap = [:]
+        filteredHostgroupConfigs.put(hostgroupSite, filteredSiteMap)
+        def index = utils.indexOfConfig(globalConfigs, hostgroupSite)
+        def globalSiteMap = index != -1 ? globalConfigs.get(index)."$hostgroupSite" : [:]
+        if (globalSiteMap.properties) {
+          globalSiteMap = globalSiteMap.properties
+        }
+        siteMap.each {
```

Comment:
`filteredSiteMap.putAll(siteMap.findAll { !globalSiteMap.containsKey(it.key) })` is shorter